app/models/view/report_date_picker.rb

```diff
@@ -0,0 +1,17 @@
+module View
+  class ReportDatePicker
+    attr_reader :formatter, :is_selected
+
+    def initialize(date, selected_date)
+      @is_selected = date == selected_date
+      @formatter = DateTimeFormatter.new(date)
+    end
+
+    delegate :database_date, :full_month_us_date,
+      to: :formatter
```

Comment:
This is from Rails. The Ruby only version is a little less nice :( Sometimes I use it though to avoid having to load all of Rails into my test suite. I am not worrying about fast tests with this app though since it is pretty small